### ISSUE TITLE
Strange error message

### ISSUE THREAD

**Issue by CBrauer** [dotnet/machinelearning#4648]
Hello,

I just upgraded my project to the new pre-release versions of ML.NET and I got the following error message when I ran my program:

![bug](https://user-images.githubusercontent.com/1317234/72281674-ccf0a400-35ef-11ea-898e-997cf19c71f0.png)

I have added a Zip file of my program and dataset.  I hope you guys can help me find out why I'm getting this error,

Charles

[bug.zip](https://github.com/dotnet/machinelearning/files/4055108/bug.zip)

**Comment by Lynx1820** [comment]
@justinormont This appears to be an AutoML issue, could you take a look at it?

**Comment by justinormont** [comment]
This is an error in the AutoML API code. 

It appears due to the handling of `NaN` values in the AutoML code, which can occur for the `F1Score` metric that you're optimizing towards. Along with `F1`, `log-loss reduction` can also return a `NaN`.

`F1` can return a `NaN` value when both the precision and recall are zero, as `F1 = 2 * precision * recall / (precision + recall)`, which turns into `0 / 0`. Precision/recall being zero occurs for a trivial model which always guesses the majority class zero.

Two things to fix:
1. Check why AutoML isn't handling the `NaN` for the metric values
2. Consider returning `F1=0.0` instead of `NaN`  -- this is done in sklearn ([code](https://github.com/scikit-learn/scikit-learn/blob/4b82c4450b759f4efd0dbad2840739ffc416e533/sklearn/metrics/classification.py#L927-L929)), and others ([note](https://github.com/dice-group/gerbil/wiki/Precision,-Recall-and-F1-measure#dividing-by-0)) 
<br/>Our `F1` calculation code:
https://github.com/dotnet/machinelearning/blob/610ffcb67083c2e5e6e1a14884ba24b1da0384c7/src/Microsoft.ML.Data/Evaluators/BinaryClassifierEvaluator.cs#L483

**Comment by CBrauer** [comment]
Thanks Justin.  I'll change your code as your suggest.
